test/jdk/java/lang/invoke/condy/CondyNameValidationTest.java

```diff
@@ -47,38 +53,41 @@ public class CondyNameValidationTest {
     static final MethodType BSM_TYPE = methodType(Object.class, MethodHandles.Lookup.class, String.class, Object.class);
 
     @DataProvider
-    public Object[][] invalidNamesProvider() throws Exception {
-        return Stream.of("",
-                         ".",
-                         ";",
-                         "[",
-                         "/")
-                .map(e -> new Object[]{e}).toArray(Object[][]::new);
+    public Object[][] invalidNamesProvider() {
+        return Stream.of(
+                        new Object[]{"", "zero-length member name"},
+                        new Object[]{".", "Invalid member name"},
+                        new Object[]{";", "Invalid member name"},
+                        new Object[]{"[", "Invalid member name"},
+                        new Object[]{"/", "Invalid member name"}
+                )
+                .toArray(Object[][]::new);
     }
 
-    @Test(dataProvider = "invalidNamesProvider", expectedExceptions = java.lang.ClassFormatError.class)
-    public void testInvalidNames(String name) throws Exception {
-        MethodHandle mh = InstructionHelper.ldcDynamicConstant(
-                L, name, Object.class,
-                "bsm", BSM_TYPE,
-                S -> {
-                });
+    @Test(dataProvider = "invalidNamesProvider")
+    public void testInvalidNames(String name, String errMessContent) throws Exception {
+        try {
+            MethodHandle mh = InstructionHelper.ldcDynamicConstant(
+                    L, name, Object.class,
+                    "bsm", BSM_TYPE
+            );
+        } catch (IllegalArgumentException e) {
+            Assert.assertTrue(e.getMessage().contains(errMessContent));
+        }
     }
 
     @DataProvider
     public Object[][] validNamesProvider() throws Exception {
         return Stream.of("<clinit>",
-                         "<init>",
-                         "<foo>")
```

Comment:
Isn't this reducing the test cases?

Reply:
<foo> is not a valid name for dynamic constant.

Reply:
You can see `java.lang.constant.ConstantUtils.validateMemberName`

Reply:
So how is it possible that `testValidNames` has been passing with the name?
Conversion should not change the tested subjects.

Reply:
Also, the test only generate bytecode, without execution.